### PR TITLE
refactored/improved notifications (works better with transitions)

### PR DESCRIPTION
**Pull request by mktcode**
- duration can be passed as parameter
- notifications have unique id to not get mixed up in DOM when elements are removed/added
- replaced over-complicated manual countdown (`setInterval(() => (now = Date.now()), 1000)`) with simpler mechanic (actually removes items from list, which eradicates the need for the additional `v-if`).
- uses the before unused `Notification` type.

notifications

**(Relies on https://github.com/snapshot-labs/snapshot/pull/963/files#diff-c0a5ddd5b365322730802e8cb15c95c3b6e965cbd3e569c69a2f089753844cdb for fade transition.)**

### CODE REVIEW

**Comment by mktcode** [comment]
@samuveth Overlay issue is fixed. I'll set this to auto merge. Review carefully! ;)

**Review by samuveth — APPROVED** [review]
Great work!

File: src/composables/useNotifications.ts [hunk 1]
```diff
@@ -1,20 +1,30 @@
 import { ref } from 'vue';
 
 interface Notification {
+  id: number;
   message: string;
   type: string;
-  timestamp: number;
+  remove(): any;
```
Comment (midgerate) [inline src/composables/useNotifications.ts:7]: is this a new way to declare a function?

Comment (midgerate) [reply src/composables/useNotifications.ts:7]: Looks fine otherwise

Comment (mktcode) [reply src/composables/useNotifications.ts:7]: Like mentioned before, I'm not too familiar with Typescript. This is what I found in examples, like here: https://www.typescriptlang.org/docs/handbook/interfaces.html

```ts
interface ClockInterface {
  tick(): void;
}
```

Comment (midgerate) [reply src/composables/useNotifications.ts:7]: Good to know that we can define functions like this 

`tick(): void`

I always used `tick = () => void`

**Review by bonustrack — APPROVED** [review]
Awesome!

**Pull request merged**